sdk/src/helpers/createDeterministicPrivateKey.ts

```diff
@@ -0,0 +1,35 @@
+import {
+    type WalletClient,
+    type Transport,
+    type Chain,
+    type Account,
+    type ParseAccount,
+    toHex,
+    keccak256,
+    Hex,
+} from 'viem';
+
+const DERIVATION_PATH_PREFIX = 'nitrolite_state_wallet_v1_';
+
+/**
+ * Generates a deterministic private key for wallet clients based on the provided adjudicator and app addresses, and a nonce.
+ */
+export const createDeterministicPrivateKey = async (
+    walletClient: WalletClient<Transport, Chain, ParseAccount<Account>>,
+    adjudicatorAddress: string,
+    appAddress: string,
+    nonce: number = 0,
+): Promise<Hex> => {
+    // NOTE: keep in mind that this key should be used only for the respective adjudicator and app address
+    const nonceMessage = `nitrolite/${adjudicatorAddress}/${appAddress}/${walletClient.account.address}/0/${nonce}`
```

Comment:
Do we not include `chainId` by design? Right now, if this signature is compromised, then all channels cross-chain with this nonce will be compromise.